ipatests/test_integration/test_ntp_replacement.py

```diff
@@ -0,0 +1,151 @@
+                                                  % ntp_server2])
+        assert expected_msg in server_install.stderr_text
+        cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
+        assert "%s" % ntp_server1 in cmd.stdout_text
+        assert "%s" % ntp_server2 in cmd.stdout_text
+        client_install = self._install_client(extraargs=[
+            '--ntp-server=%s' % ntp_server1,
+            '--ntp-server=%s' % ntp_server2])
+        assert expected_msg in client_install.stderr_text
+        cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
+        assert "%s" % ntp_server1 in cmd.stdout_text
+        assert "%s" % ntp_server2 in cmd.stdout_text
+        self.cleanup()
+
+    def test_server_and_client_install_with_ntp_pool_and_ntp_server(self):
+        """
+        test to verify ipa-server and ipa-client install with
+        options --ntp-pool and --ntp-server together
+        """
+        ntp_pool = "pool.ntp.org"
+        ntp_server = "1.pool.ntp.org"
+        expected_msg = "Configuration of chrony was changed by installer."
+        server_install = tasks.install_master(self.master, setup_dns=False,
+                                              extra_args=[
+                                                  '--ntp-pool=%s' % ntp_pool,
+                                                  '--ntp-server=%s'
+                                                  % ntp_server])
+        assert expected_msg in server_install.stderr_text
+        cmd = self.master.run_command(['cat', paths.CHRONY_CONF])
+        assert "%s" % ntp_pool in cmd.stdout_text
+        assert "%s" % ntp_server in cmd.stdout_text
+        client_install = self._install_client(extraargs=[
+            '--ntp-pool=%s' % ntp_pool,
+            '--ntp-server=%s' % ntp_server])
+        assert expected_msg in client_install.stderr_text
+        cmd = self.client.run_command(['cat', paths.CHRONY_CONF])
+        assert "%s" % ntp_pool in cmd.stdout_text
+        assert "%s" % ntp_server in cmd.stdout_text
+        self.cleanup()
+
+    def test_server_and_client_install_with_mixed_options(self):
+        """
+        test to verify ipa-server and ipa-client install with
+         --ntp-server and -N options
+        """
+        ntp_server = "1.pool.ntp.org"
+        exp_str = ("error: --ntp-server cannot be used"
+                   " together with --no-ntp")
+        args1 = ['ipa-server-install', '-N', '--ntp-server=%s' % ntp_server]
+        server_install = self.master.run_command(args1, raiseonerr=False)
+        assert server_install.returncode == 2
+        assert exp_str in server_install.stderr_text
+        args2 = ['ipa-client-install', '-N', '--ntp-server=%s' % ntp_server]
+        client_install = self.master.run_command(args2, raiseonerr=False)
+        assert client_install.returncode == 2
+        assert exp_str in client_install.stderr_text
+
+    def test_server_install_with_dns_setup_and_ntp_server(self):
```

Comment:
Why do we need another test scenario for DNS + NTP?

Reply:
I agree this scenario is not required but it could be prevented regressions.

Reply:
IMHO NTP service should not affect DNS setup, thus this scenario is not needed.
Would not be better to try replica installation?

Reply:
IMHO NTP service should not affect DNS setup, thus this scenario is not needed.
Would not be better to try replica installation?

Reply:
@varunmylaraiah, please make it as a part of this one :)